unitylibs/core/workflow/workflow-acrobat/action-binder.js

```diff
@@ -0,0 +1,151 @@
+      // return;
+    }
+    const file = files[0];
+    if (!file) return;
+    const MAX_FILE_SIZE = 1000000000;
+    if (file.size > MAX_FILE_SIZE) {
+      // unityEl.dispatchEvent(new CustomEvent(errorToastEvent, { detail: { msg: File too large. } }));
+      // return;
+    }
+    if (file.type != 'application/pdf') {
+      // unityEl.dispatchEvent(new CustomEvent(errorToastEvent, { detail: { msg: 'Unsupported file type' } }));
+      // return;
+    }
+    // Create asset
+    const assetData = await this.serviceHandler.postCallToService(
+      this.acrobatApiConfig.acrobatEndpoint.createAsset,
+      {
+        'surfaceId': 'acrobat',
+        'name': file.name,
+        'size': file.size,
+        'format': file.type
+      }
+    );
+    if (assetData?.status !== 200) {
+      // unityEl.dispatchEvent(new CustomEvent(errorToastEvent, { detail: { msg: 'Unable to process the request' } }));
+      // return;
+    }
+    // Chunk PDF and upload
+    const pdfUpload = chunkPdf(assetData);
+    if (pdfUpload?.status !== 200) {
+      // unityEl.dispatchEvent(new CustomEvent(errorToastEvent, { detail: { msg: 'Unable to process the request' } }));
+      // return;
+    }
+    // Finalize asset
+    const finalizeAsset = await this.serviceHandler.postCallToService(
+      this.acrobatApiConfig.acrobatEndpoint.finalizeAsset,
+      {
+        'surfaceId': 'acrobat',
+        'assetId': assetData.id
+      }
+    );
+    if (finalizeAsset?.status !== 200) {
+      // unityEl.dispatchEvent(new CustomEvent(errorToastEvent, { detail: { msg: 'Unable to process the request' } }));
+      // return;
+    }
+    // Redirect to Acrobat Product
+    // continueInApp();
+  }
+
+  // TODO: Customize this function for acrobat use case
+  // continueInApp() {
+  //   const cOpts = {
+  //     assetId: null,
+  //     targetProduct: 'Photoshop',
+  //     payload: {
+  //       finalAssetId: null,
+  //       operations: [],
+  //     },
+  //   };
+  //   this.operations.forEach((op, i) => {
+  //     const idx = cOpts.payload.operations.length;
+  //     if ((i > 0) && (this.operations[i - 1].operationType == op.operationType)) {
+  //       cOpts.payload.operations[idx - 1][op.adjustmentType] = parseInt(op.filterValue.sliderElem.value, 10);
+  //     } else {
+  //       cOpts.payload.operations.push({ name: op.operationType });
+  //       if (op.assetId && !cOpts.assetId) cOpts.assetId = op.assetId;
+  //       if (op.assetId) cOpts.payload.finalAssetId = op.assetId;
+  //       if (op.operationType == 'changeBackground') cOpts.payload.operations[idx].assetIds = [op.assetId];
+  //       if (op.adjustmentType && op.filterValue) {
+  //         cOpts.payload.operations[idx][op.adjustmentType] = parseInt(op.filterValue.sliderElem.value, 10);
+  //       }
+  //     }
+  //   });
+  //   this.serviceHandler.postCallToService(
+  //     this.psApiConfig.connectorApiEndPoint,
+  //     { body: JSON.stringify(cOpts) },
+  //   );
+  // }
+}
```

Comment:
As you mentined in the comments on slack we would need cleanup and anyways this will need more development to integrate with backend service.
But can we remove the comments to make this more readable?